src/v2/Apps/Settings/Routes/Payments/Components/SettingsPaymentsMethod.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  Clickable,
+  CreditCardIcon,
+  Flex,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import { FC, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { SettingsPaymentsMethod_method } from "v2/__generated__/SettingsPaymentsMethod_method.graphql"
+import { useDeleteCreditCard } from "../useDeleteCreditCard"
+
+enum Mode {
+  Pending,
+  Deleting,
+}
+
+interface SettingsPaymentsMethodProps {
+  method: SettingsPaymentsMethod_method
+}
+
+const SettingsPaymentsMethod: FC<SettingsPaymentsMethodProps> = ({
+  method,
+}) => {
+  const [mode, setMode] = useState(Mode.Pending)
```

Comment:
We use this pattern in so many spots now -- its a good pattern -- but starting to wonder if we might not want to migrate it to a hook? 

```tsx
const [mode, setMode, Pending, Deleting] = useMode(['pending', 'deleting']) 
```

Reply:
I was also wondering about this but couldn't come up with an API that was more succinct than just typing it out manually. Lemme see if I can make this example API work.

Reply:
every PR with this pattern I wonder the same thing, and then assume ^, but i keep on wondering about it so... maybe there's something there

Reply:
Hm, yeah; really there's not a great way to do it type-wise without casting that array `as const`.

Really the most compact expression of this is simply to use strings, but specify the valid type:

```tsx
const [mode, setMode] = useState<'Pending' | 'Saving'>('Pending')
```

Reply:
Maybe I can look at this today real quick